src/app/Trade.ts

```diff
@@ -38,47 +38,81 @@ export class Trade {
   }
 
   private areOffersInvalid(
-    [animal, count]: Offer,
+    offer: Offer[],
     playerHerd: Herd,
-    [bankAnimal, bankCount]: Offer,
+    target: Offer[],
   ): boolean {
     return (
-      (count > 1 && bankCount > 1) ||
-      playerHerd.getAnimalNumber(animal) < count ||
-      this.bank.theHerd.getAnimalNumber(bankAnimal) < bankCount
+      (offer.length > 1 && target.length > 1) ||
+      this.isOfferQuantityInvalid(offer, playerHerd) ||
+      this.isOfferQuantityInvalid(target, this.bank.theHerd) ||
+      !this.isSingleQuantityOnAnySide(offer, target)
     );
   }
 
-  private calculateValue([animal, count]: Offer): number {
-    return multiply(
-      ConvertAnimalName.toAnimalObject(animal).theValue,
-      count,
+  private isOfferQuantityInvalid(
+    offer: Offer[],
+    herd: Herd,
+  ): boolean {
+    return (
+      offer.filter(
+        ([animal, count]) => herd.getAnimalNumber(animal) >= count,
+      ).length !== offer.length
+    );
+  }
+
+  private isSingleQuantityOnAnySide(
+    offer: Offer[],
+    target: Offer[],
+  ): boolean {
+    return (
+      (offer.length === 1 && offer[0][1] === 1) ||
+      (target.length === 1 && target[0][1] === 1)
     );
   }
 
-  private adjustOffer(offer: Offer, target: Offer): void {
-    offer[1] -= 1;
+  private calculateValue(offer: Offer[]): number {
+    return sum(
+      offer.map(([animal, count]) =>
+        multiply(
+          ConvertAnimalName.toAnimalObject(animal).theValue,
+          count,
+        ),
+      ),
+    );
+  }
+
+  private adjustOffer(offer: Offer[], target: Offer[]): void {
+    this.removeQuantityOfOffer(offer);
     if (this.calculateValue(offer) <= this.calculateValue(target)) {
       return;
     }
     this.adjustOffer(offer, target);
   }
 
+  private removeQuantityOfOffer(offer: Offer[]): void {
```

Comment:
These lines are unclear what they do. Add a description please.

Reply:
I have removed that method and replaced this with more suitable solution